daemon/config/config_common_unix_test.go

```diff
@@ -41,3 +41,29 @@ func TestCommonUnixValidateConfigurationErrors(t *testing.T) {
 		}
 	}
 }
+
+func TestCommonUnixGetInitPath(t *testing.T) {
+	testCases := []struct {
+		config           *Config
+		expectedInitPath string
+	}{
+		{
+			config: &Config{
+				CommonUnixConfig: CommonUnixConfig{
+					DefaultInitBinary: "foo-init-bin",
+				},
+			},
+			expectedInitPath: "foo-init-bin",
+		},
+		{
+			config:           &Config{},
+			expectedInitPath: "docker-init",
+		},
```

Comment:
@thaJeztah lol! that's alright 😁 Added another test-case and squashed the commits.